subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -269,6 +269,58 @@ eclipse {
 Previously, you had to explicitly configure PMD to ignore default rules with `ruleSets = []`.
 In the Gradle 8.0, setting `ruleSetConfig` or `ruleSetFiles` to a non-empty value implicitly ignores default rules.
 
+==== Changing test framework after setting test framework options is now an error
+
+When configuring the built-in test task for Java, Groovy, and Scala projects, Gradle no longer allows you to
+change the test framework used by the `Test` task after configuring options. This was deprecated since it silently
+discarded configuration in some cases.
+
+The following code example now produces an error:
+```
+test {
+   options {
+   }
+
+   useJUnitPlatform()
+}
+```
+Instead, you can:
+
+- set the test framework before configuring options
+- migrate to the <<jvm_test_suite_plugin#jvm_test_suite_plugin,JVM Test Suite Plugin>>
+```
+test {
+   // select test framework before configuring options
+   useJUnitPlatform()
+   options {
+   }
+}
+```
+
+==== Report `getOutputLocation` return type changed
```

Comment:
```suggestion
==== Report `getOutputLocation` return type changed from Provider to Property
```
Suggestion: make the section title more descriptive, provide a hint to readers since non-native English speakers especially might have trouble seeing this change inline at first glance